src/Collection/CollectionTrait.php

```diff
@@ -35,13 +35,27 @@
 use Traversable;
 
 /**
- * Offers a handful of method to manipulate iterators
+ * Offers a handful of methods to manipulate iterators
  */
 trait CollectionTrait
 {
 
     use ExtractTrait;
 
+    /**
+     * Returns a new collection.
+     *
+     * Allows classes which use this trait to determine their own
+     * type of returned collection interface
+     *
+     * @param mixed $args,.. Constructor arguments.
+     * @return \Cake\Collection\CollectionInterface
+     */
+    protected function newCollection(...$args)
```

Comment:
Why not just using the same typehint as the collection constructor?

    public function __construct(iterable $items)

Reply:
Because child classes of `Collection` can have different constructor signatures. For e.g. `ReplaceIterator` has `__construct($items, callable $callback)`.